op-acceptor/logging/filelogger.go

```diff
@@ -0,0 +1,249 @@
+package logging
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ethereum-optimism/infra/op-acceptor/types"
+)
+
+// FileLogger handles writing test output to files
+type FileLogger struct {
+	baseDir     string // Base directory for logs
+	logDir      string // Root log directory
+	failedDir   string
+	summaryFile string
+	mu          sync.Mutex // Protects concurrent file operations
+}
+
+// NewFileLogger creates a new file logger with the specified base directory and runID
+// The runID must be provided, otherwise an error is returned
+func NewFileLogger(baseDir string, runID string) (*FileLogger, error) {
+	if runID == "" {
+		return nil, fmt.Errorf("runID is required")
+	}
+
+	dirName := fmt.Sprintf("testrun-%s", runID)
+	runDir := filepath.Join(baseDir, dirName)
+	failDir := filepath.Join(runDir, "failed")
+
+	// Create directories
+	dirs := []string{
+		runDir,
+		filepath.Join(runDir, "tests"),
+		failDir,
+	}
+
+	for _, dir := range dirs {
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			return nil, fmt.Errorf("failed to create directory %s: %w", dir, err)
+		}
+	}
+
+	return &FileLogger{
+		baseDir:     runDir,
+		logDir:      baseDir,
+		failedDir:   failDir,
+		summaryFile: filepath.Join(runDir, "summary.log"),
+	}, nil
+}
+
+// GetDirectoryForRunID returns the path for a specific runID
+// The runID must be provided, otherwise an error is returned
+func (l *FileLogger) GetDirectoryForRunID(runID string) (string, error) {
```

Comment:
The `NewFileLogger` accepts a `runID`, why not store it on the struct?

Reply:
The FileLogger can persist across multiple runs. So I didn't persist it simply to reduce the chance that the wrong ID would be used.